egui_extras/src/sizing.rs

```diff
@@ -0,0 +1,85 @@
+/// Size hint for table column/grid cell
+#[derive(Clone, Debug)]
+pub enum Size {
+    /// Absolute size in points
+    Absolute(f32),
+    /// Relative size relative to all available space. Values must be in range `0.0..=1.0`
+    Relative(f32),
+    /// [`Size::Relative`] with a minimum size in points
+    RelativeMinimum {
+        /// Relative size relative to all available space. Values must be in range `0.0..=1.0`
+        relative: f32,
+        /// Absolute minimum size in points
+        minimum: f32,
+    },
+    /// Multiple remainders each get the same space
+    Remainder,
+    ///  [`Size::Remainder`] with a minimum size in points
+    RemainderMinimum(f32),
+}
+
+pub struct Sizing {
+    sizes: Vec<Size>,
+}
+
+impl Sizing {
+    pub fn new() -> Self {
+        Self { sizes: vec![] }
+    }
+
+    pub fn add_size(&mut self, size: Size) {
```

Comment:
```suggestion
    pub fn add(&mut self, size: Size) {
```

it's the only thing you can add 🤷  :)